contracts/core/mpl-rewards/v1/test/helper/Rewards.sol

```diff
@@ -0,0 +1,740 @@
+
+        assertTrue(!fay.try_stake(stakeAmt * WAD));  // Can't stake before approval
+
+        fay.increaseCustodyAllowance(address(mplRewards), stakeAmt * WAD);
+
+        assertTrue(!fay.try_stake(0));          // Can't stake zero
+        assertTrue( fay.try_stake(stakeAmt * WAD));  // Can stake after approval
+
+        assertEq(stakeToken.balanceOf(address(fay)),     amt);  // PoolFDT balance doesn't change
+        assertEq(mplRewards.balanceOf(address(fay)),     100 * WAD);
+        assertEq(mplRewards.totalSupply(),               100 * WAD);
+
+        checkDepositOrStakeDate(isPfdtStakeToken, start, stakeToken, fay);
+    }
+
+    function withdraw_test(bool isPfdtStakeToken, uint256 amt, uint256 stakeAmt, IStakeToken stakeToken) public {
+        uint256 start = block.timestamp;
+
+        if (isPfdtStakeToken) {
+            mintFundsAndDepositIntoPool(fay, pool1, amt * USD, amt * USD);
+            assertEq(stakeToken.balanceOf(address(fay)), amt * WAD);
+            pat.setLockupPeriod(address(pool1), 0);
+        } else {
+            setUpForStakeLocker(amt, sam, fay);
+        }
+
+        amt      *= WAD;
+        stakeAmt *= WAD;
+
+        fay.increaseCustodyAllowance(address(mplRewards), stakeAmt);
+        fay.stake(stakeAmt);
+
+        hevm.warp(start + 1 days);  // Warp to ensure no effect on depositDates
+
+        checkDepositOrStakeDate(isPfdtStakeToken, start, stakeToken, fay);
+
+        assertEq(stakeToken.balanceOf(address(fay)),      amt);  // FDT balance doesn't change
+        assertEq(mplRewards.balanceOf(address(fay)), stakeAmt);
+        assertEq(mplRewards.totalSupply(),           stakeAmt);
+
+        assertTrue(!fay.try_withdraw(0));         // Can't withdraw zero
+        assertTrue( fay.try_withdraw(stakeAmt));  // Can withdraw
+
+        checkDepositOrStakeDate(isPfdtStakeToken, start, stakeToken, fay);
+
+        assertEq(stakeToken.balanceOf(address(fay)), amt);
+        assertEq(mplRewards.balanceOf(address(fay)),   0);
+        assertEq(mplRewards.totalSupply(),             0);
+    }
+
+    /**********************************/
+    /*** Rewards accounting testing ***/
+    /**********************************/
+
+    function rewards_single_epoch_test(bool isPfdtStakeToken, uint256 amt, IStakeToken stakeToken) public {
+
+        if (isPfdtStakeToken) {
+            mintFundsAndDepositIntoPool(fay, pool1, amt * USD, amt * USD);
+            mintFundsAndDepositIntoPool(fez, pool1, amt * USD, amt * USD);
+            pat.setLockupPeriod(address(pool1), 0);
+        } else {
+            mint("BPT", address(sam), 100 * WAD);
+            mint("BPT", address(sid), 100 * WAD);
+            setUpForStakeLocker(amt, sam, fay);
+            setUpForStakeLocker(amt, sid, fez);
+        }
+
+        fay.increaseCustodyAllowance(address(mplRewards), 100 * WAD);
+        fez.increaseCustodyAllowance(address(mplRewards), 100 * WAD);
```

Comment:
Lets express this entire test in terms of `amt`

Reply:
If we want it then it will lead to a lot of mathematical changes as we are assuming that its funds are multiple of 10. or it may be just of rework to make it agnostic